src/CSAccounting.sol

```diff
@@ -632,19 +652,25 @@ contract CSAccounting is
             cumulativeFeeShares,
             rewardsProof
         );
-
+        // NOTE: Increase the bond shares first to respect the invariant:
+        //       any feature amount to transfer from bond should be bounded by `claimable`.
+        CSBondCore._increaseBond(nodeOperatorId, distributed);
```

Comment:
I suggest modifying the getRequiredBond function so that it will accept additional bonds and use them. The current approach does not look optimal from the storage usage perspective